### PR TITLE
[TRTLLM-13052][feat] Enable TRTLLM moe backend for nemotron-h BF16 ckpt

### PR DESCRIPTION
**Pull request by Wanli-Jiang**
## Summary by CodeRabbit

* **New Features**
  * MOE backend now supports multiple activation types (Relu2, SwiGLU) in BF16 unquantized FlashInfer path.

* **Improvements**
  * Simplified MOE class fallback logic for unquantized configurations.
  * Enhanced routing and backend configuration validation.

* **Tests**
  * Added comprehensive accuracy tests for BF16 TRTLLM MOE backend across multiple configurations and hardware platforms.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Review by rosenrodt — APPROVED** [review]
LGTM. Thanks.

File: tests/integration/defs/accuracy/test_llm_api_pytorch.py [hunk 1]
```diff
@@ -6816,6 +6816,35 @@ def test_auto_dtype_4gpus(self, tp_size, ep_size, attention_dp,
...
+        with LLM(
+                f"{llm_models_root()}/NVIDIA-Nemotron-3-Super-120B-A12B-BF16",
+                kv_cache_config=kv_cache_config,
+                max_batch_size=32,
+                tensor_parallel_size=4,
+                moe_expert_parallel_size=4,
+                enable_attention_dp=attention_dp,
+                moe_config=MoeConfig(backend="TRTLLM"),
+                **pytorch_config,
+        ) as llm:
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply tests/integration/defs/accuracy/test_llm_api_pytorch.py:6840]: it will guard by skip_preblackwell and the memory constraint.

**Review by nv-guomingz — APPROVED** [review]

File: tests/unittest/_torch/modules/moe/test_moe_backend.py [hunk 2]
```diff
@@ -744,3 +747,131 @@ def run_moe():
...
+
+@pytest.mark.parametrize(
+    "trtllm_use_router_logits", [True, False], ids=["fused_routing", "separated_routing"]
+)
+@pytest.mark.parametrize("seq_len", [8, 256])
+@pytest.mark.parametrize(
+    "activation_type", [ActivationType.Relu2, ActivationType.Swiglu], ids=["relu2", "swiglu"]
+)
+@pytest.mark.parametrize("routing_kind", ["deepseekv3", "renormalize"])
+def test_trtllm_bf16_unquantized_moe(
```
Comment (yuxianq) [inline tests/unittest/_torch/modules/moe/test_moe_backend.py:788]: Why we add a new `test_trtllm_bf16_unquantized_moe` instead of adding cases to `test_moe_backend`?

Comment (rosenrodt) [reply tests/unittest/_torch/modules/moe/test_moe_backend.py:788]: It makes sense to add sub case to test_moe_backend. But IIUC all test_moe_backend test cases containing "TRTLLM" are currently blanket waived when the intent was only waiving certain sub cases. It may not be the MR's goal to untangle the whole thing.

Comment (yuxianq) [reply tests/unittest/_torch/modules/moe/test_moe_backend.py:788]: Make sense. Thanks for explaining the background.

**Review by yuxianq — APPROVED** [review]

File: tests/unittest/_torch/modules/moe/test_moe_backend.py (hunk 2 above)
Comment (Wanli-Jiang) [reply tests/unittest/_torch/modules/moe/test_moe_backend.py:788]: actually some tests under `test_moe_backend` were failed. I will keep use the separated tests to make the testing cleaner!

**Review by xinhe-nv — APPROVED** [review]

**Comment by Wanli-Jiang** [comment]
/bot skip --comment "Test passed at 52634 for single GPUs and 52923 for multi GPUs"
